accounting/account/models.py

```diff
@@ -89,6 +89,14 @@ def name(self):
         """
         return self.business_name or self.user.get_full_name() or self.user.username
 
+    @property
+    def active_bank_account(self):
+        """
+        Return the user's currently active bank account.
+        """
+        # TODO: Make it so only one bank account can be 'active' at a time.
```

Comment:
If we associate a bank account with an invoice, that would read that an invoice belongs to a bank account, which doesn't sound like the correct relation to me. And both the client and provider on the invoice can have bank accounts.

If we associate it with an hourly rate, the latter problem again applies: a bank account for whom, the client or provider?

Reply:
For a concrete future plan, I would think using something similar to edX's configuration model where you can set an object as 'active' would be the way to go. Or just change this relation to be 1-to-1 between bank account & user account.

Reply:
@UmanShahzad 

> ... Or just change this relation to be 1-to-1 between bank account & user account.

That seems like a good solution for now. In the future it would be cool for clients/providers to be able to select a bank account to use for a specific invoice. But unless we have another concrete use case that we need to satisfy now (?), I'd be in favor of simplifying things and moving to a 1-to-1 relation as you suggest.

Reply:
Yeah, no use case to satisfy right now. But since we're already using this property to encapsulate the issue here, it would be more work to undo it, make it 1-to-1, and then in the future have to redo it by doing another migration from 1-to-1 to 1-to-many, etc.. So for the sake of time, and because this feature could be useful, I'd not remove it right now lest it lead to more unnecessary work.

Reply:
@UmanShahzad With all of the automation work we have coming up I have a feeling that this code will stay around for quite a while... Even though we might not implement a nice-to-have such as being able to select a bank account for a specific invoice in the near future.

But OK, let's wrap this up now -- we can see if Xavier has any feedback on this during upstream review, and make changes based on that (if necessary) :)

Reply:
@itsjeyd Yeah that's a good point. I guess what I'm getting at is, at this point it's going to be more work to undo this and go to 1-to-1.

@antoviaque I think you'll probably want this to be 1-to-1 as well, but see my above comments: it's going to delay merging this PR and eat up time; it's working fine as is but yes the design could in principle be changed to match current use cases.

Reply:
@itsjeyd @UmanShahzad Yup - I actually highly hesitated to point this out in the initial review, but then I thought I was already being quite a pain. Generally I'm in favor to keeping things simpler and closer to the existing use cases, rather than trying to plan ahead for things we might need one day. Usually that need never turns out to be needed, or not the way we initially thought, and we are just stuck with unnecessary complexity. So if it fits into the time you have allocated for addressing reviews this week, I would also be in favor of going towards the 1-to-1 relationship.

Reply:
@antoviaque At this point I've gravitated towards that mindset too. Let's see if I have extra time available for this. Between having to write tests and this though, I will be prioritizing investing effort on tests.

Reply:
Created https://tasks.opencraft.com/browse/OC-3702.